### DISCUSSION TITLE
Multiple Output Options, Per Run.

### DISCUSSION THREAD

**Discussion by di-org** [projectdiscovery/subfinder#454] [Ideas]
1st, really appreciate the container build.

If this isn't too much of an edge case.... I find it convenient to have both just the full "-o FILE" output (for further analysis) as well as the "-nW -oI -o FILE" output when doing external engagements.  

I have been taking the "-o" output and running against "nmap -sL" to find the hosts that resolve, but in large domains, and compared to the speed of Subfinder's -oI option, nmap is quite slow.  I assume it is bad form to run them back to back, just to get different outputs.

Would it be feasible to output both formats on a single run, or to be able to take a file of subdomains and have a switch to just run the resolution portion w/ and use the -o to output the results?.

Thanks,
Andy

**Comment by ehsandeep** [comment]
You can do the following without making any change @di-org, you can utilize subfinder for passive enumeration and [dsnx](https://github.com/projectdiscovery/dnsx) to run the resolution, subfinder uses dnsx internally for resolution, but in this way, you can have both output, if you looking for more data for resolved host, you can also use `-json` flag of dnsx.

```
subfinder -d xx.com -o passive | dnsx -o active
```

**Reply by di-org** [reply]
I'll give that a try.  Thanks!

**Reply by di-org** [reply]
Hey @eshandeep ,  I'm trying to run both subfinder and dnsx as containers.  I am not sure if I'm missing something, as I'm fairly new at this.

I've used $PWD:/root/.config/subfinder to write the file out, and tried to use "$PWD:/root/.config/dnsx as well as subfinder to read the file into dnsx with -f.  Is it possible to run both in separate containers?  Sorry for the newb questions.
